### PR TITLE
chore(mint-client): deprecate redundant `get_notes_tier_counts` function

### PR DESCRIPTION
**Pull request by elsirion**

### CODE REVIEW

File: modules/fedimint-mint-client/src/lib.rs [hunk 1]
```diff
@@ -1012,17 +1012,9 @@ impl MintClientModule {
...
-        dbtx.find_by_prefix(&NoteKeyPrefix)
-            .await
-            .fold(
-                TieredCounts::default(),
-                |mut acc, (key, _note)| async move {
-                    acc.inc(key.amount, 1);
-                    acc
-                },
-            )
-            .await
```
Comment (elsirion) [inline modules/fedimint-mint-client/src/lib.rs:1025]: My IDE warned me about duplicate code while reading the module

**Review by dpc — APPROVED** [review]
I would keep `get_notes_tier_counts`, as it is just more concrete, while "wallet summary" doesn't explain much.

**Comment by elsirion** [comment]
@joschisan @alexlwn123 @maan2003 thoughts? (just people I remember having opinions on naming before)

**Comment by alexlwn123** [comment]
maybe `get_notes_breakdown` or `get_notes_denomination_counts` or `get_note_counts_by_denomination`?

**Comment by elsirion** [comment]
I like `get_note_counts_by_denomination` a lot! Thoughts @dpc?

**Comment by dpc** [comment]
> I like `get_note_counts_by_denomination` a lot! Thoughts @dpc?

Beautiful.

**Review by dpc — APPROVED** [review]

**Pull request merged**